crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -2332,6 +2332,189 @@ impl<F: Clone> PostUpdateTracker<F, PaymentConfirmData<F>, types::PaymentsAuthor
     }
 }
 
+#[cfg(feature = "v2")]
+impl<F: Send + Clone> Operation<F, types::PaymentsSyncData> for PaymentResponse {
+    type Data = PaymentStatusData<F>;
+    fn to_post_update_tracker(
+        &self,
+    ) -> RouterResult<&(dyn PostUpdateTracker<F, Self::Data, types::PaymentsSyncData> + Send + Sync)>
+    {
+        Ok(self)
+    }
+}
+
+#[cfg(feature = "v2")]
+#[async_trait]
+impl<F: Clone> PostUpdateTracker<F, PaymentStatusData<F>, types::PaymentsSyncData>
+    for PaymentResponse
+{
+    async fn update_tracker<'b>(
+        &'b self,
+        state: &'b SessionState,
+        mut payment_data: PaymentStatusData<F>,
+        response: types::RouterData<F, types::PaymentsSyncData, types::PaymentsResponseData>,
+        key_store: &domain::MerchantKeyStore,
+        storage_scheme: enums::MerchantStorageScheme,
+        locale: &Option<String>,
+        #[cfg(all(feature = "v1", feature = "dynamic_routing"))] routable_connector: Vec<
+            RoutableConnectorChoice,
+        >,
+        #[cfg(all(feature = "v1", feature = "dynamic_routing"))] business_profile: &domain::Profile,
+    ) -> RouterResult<PaymentStatusData<F>>
+    where
+        F: 'b + Send + Sync,
+    {
+        use common_utils::ext_traits::OptionExt;
+
+        let db = &*state.store;
+        let key_manager_state = &state.into();
+
+        let response_router_data = response;
+        let payment_attempt = payment_data
+            .payment_attempt
+            .clone()
+            .get_required_value("payment_attempt")
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("payment attempt is not set in payment data")?;
+
+        match response_router_data.response {
+            Ok(response) => match response {
+                types::PaymentsResponseData::TransactionResponse {
+                    resource_id,
+                    redirection_data,
+                    mandate_reference,
+                    connector_metadata,
+                    network_txn_id,
+                    connector_response_reference_id,
+                    incremental_authorization_allowed,
+                    charge_id,
+                } => {
+                    let attempt_status = response_router_data.status;
+                    let intent_status = common_enums::IntentStatus::foreign_from(attempt_status);
+                    let connector_payment_id = match resource_id {
+                        types::ResponseId::NoResponseId => None,
+                        types::ResponseId::ConnectorTransactionId(id)
+                        | types::ResponseId::EncodedData(id) => Some(id),
+                    };
+
+                    let payment_intent_update = hyperswitch_domain_models::payments::payment_intent::PaymentIntentUpdate::SyncUpdate { status: intent_status, updated_by: storage_scheme.to_string() };
+                    let updated_payment_intent = db
```

Comment:
We should not do these, branch based update for each variant, do the conversion first then call the update.